In [1]:
%cd /content
!rm -rf Yaqeen
!git clone -q -b extension-mv3 https://github.com/sabascodes/Yaqeen.git
%cd /content/Yaqeen/backend
print("✓ تم إحضار الكود")

/content
/content/Yaqeen/backend
✓ تم إحضار الكود


In [2]:
!pip install -q -r requirements-dev.txt -r requirements-semantic.txt
print("✓ تم التثبيت")

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 102.9 MB/s eta 0:00:00
✓ تم التثبيت


In [3]:
!python -m pytest -q

.......................                                                  [100%]
23 passed in 1.43s


In [4]:
!python fetch_quran.py

سورة 114/114
حُفظت 6236 آية في /content/Yaqeen/backend/data/quran.json


In [5]:
!python build_hadith.py

عدد الأحاديث: 3574
3574/3574
حُفظت في /content/Yaqeen/backend/data/hadeethenc.json
modules.json: 100% 387/387 [00:00<00:00, 1.41MB/s]
README.md: 100% 179k/179k [00:00<00:00, 94.0MB/s]
sentence_bert_config.json: 100% 57.0/57.0 [00:00<00:00, 258kB/s]
config.json: 100% 694/694 [00:00<00:00, 3.60MB/s]

model.safetensors: downloading bytes:  20% 225M/1.11G [00:02<00:05, 160MB/s, 19.2MB/s  ]
model.safetensors: reconstructing file:  23% 256M/1.11G [00:02<00:07, 120MB/s]
model.safetensors: downloading bytes:  66% 739M/1.11G [00:03<00:01, 279MB/s, 58.5MB/s  ]
model.safetensors: reconstructing file:  48% 536M/1.11G [00:04<00:04, 124MB/s, 25.1MB/s  ]
model.safetensors: downloading bytes: 100% 741M/741M [00:15<00:00, 48.6MB/s, 60.6MB/s  ]
model.safetensors: reconstructing file: 100% 1.11G/1.11G [00:15<00:00, 72.9MB/s, 62.7MB/s  ]
Loading weights: 100% 199/199 [00:00<00:00, 5447.57it/s]
tokenizer_config.json: 100% 418/418 [00:00<00:00, 1.55MB/s]

sentencepiece.bpe.model: downloading bytes:  37% 1.8

In [6]:
النص = "قال رسول الله ﷺ: إنما الأعمال بالنيات وإنما لكل امرئ ما نوى"  #@param {type:"string"}

import os, sys
os.chdir("/content/Yaqeen/backend"); sys.path.insert(0, ".")
from fastapi.testclient import TestClient
import main
r = TestClient(main.app).post("/verify", json={"text": النص}).json()
if r.get("status") == "error":
    print("تعذر الوصول إلى الدرر السنية الآن، جرّبي مرة أخرى.")
card = r.get("card", r)
for k, v in card.items():
    if v:
        print(f"{k}: {v}")

Loading weights:   0%|          | 0/199 [00:00<?, ?it/s]

type: wording
detected: قال رسول الله ﷺ: إنما الأعمال بالنيات وإنما لكل امرئ ما نوى
source: موسوعة الأحاديث النبوية · متفق عليه
ruling: صحيح · متفق عليه
correct: عن عمر بن الخطاب رضي الله عنه قال: قال رسول الله صلى الله عليه وسلم: «إِنَّمَا الْأَعْمَالُ بِالنِّيَّةِ، وَإِنَّمَا لِامْرِئٍ مَا نَوَى، فَمَنْ كَانَتْ هِجْرَتُهُ إِلَى اللهِ وَرَسُولِهِ، فَهِجْرَتُهُ إِلَى اللهِ وَرَسُولِهِ، وَمَنْ كَانَتْ هِجْرَتُهُ لِدُنْيَا يُصِيبُهَا أَوِ امْرَأَةٍ يَتَزَوَّجُهَا، فَهِجْرَتُهُ إِلَى مَا هَاجَرَ إِلَيْهِ».
وفي لفظ للبخاري: «إِنَّمَا الْأَعْمَالُ بِالنِّيَّاتِ، وَإِنَّمَا لِكُلِّ امْرِئٍ مَا نَوَى».
note: الحكم والتخريج كما نشرتهما موسوعة الأحاديث النبوية.
confidence: high
url: https://hadeethenc.com/ar/browse/hadith/4560


In [7]:
from fastapi.testclient import TestClient
import main
for k, v in TestClient(main.app).get("/").json().items():
    print(f"{k}: {v}")

app: yaqeen
ok: True
quran_ayat: 6236
hadeethenc_hadith: 3574
semantic: True
semantic_model: intfloat/multilingual-e5-base
hadith_semantic: True
hadith_semantic_model: intfloat/multilingual-e5-base


In [8]:
%cd /content/Yaqeen/extension
!curl -fsSL https://deb.nodesource.com/setup_22.x | bash - > /dev/null 2>&1
!apt-get install -y -qq nodejs > /dev/null 2>&1
!node -v
!npm install --ignore-scripts --silent
!npm test
!npm run build
!cd dist && zip -qr /content/yaqeen-extension.zip .
from google.colab import files
files.download("/content/yaqeen-extension.zip")

/content/Yaqeen/extension
v22.23.3

> yaqeen@0.1.0 test
> vitest run


 RUN  v5.0.3 /content/Yaqeen/extension


 ❯ test/core.test.ts [queued]

 Test Files 0 passed (1)
      Tests 0 passed (0)
   Start at 10:40:17
   Duration 101ms

 ❯ test/core.test.ts 0/41

 Test Files 0 passed (1)
      Tests 0 passed (41)
   Start at 10:40:17
   Duration 452ms
 ✓ test/core.test.ts (41 tests) 77ms
   ✓ normalization (1)
     ✓ makes Uthmani and everyday spelling comparable 5ms
   ✓ Quran matching (7)
     ✓ matches an exact quote with surah and ayah 21ms
     ✓ flags a misquoted ayah and shows the correct wording 2ms
     ✓ matches part of an ayah 1ms
     ✓ matches a quote spanning consecutive ayat 1ms
     ✓ finds an ayah inside a longer post 3ms
     ✓ reports other places of a repeated ayah 1ms
     ✓ ignores ordinary text 2ms
   ✓ hadith (7)
     ✓ treats a one-word change as different wording 2ms
     ✓ matches an authentic hadith from HadeethEnc locally 2ms
     ✓ flags different wording of a

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [9]:
%cd /content/Yaqeen/backend
!zip -qj /content/yaqeen-data.zip data/quran.json data/hadeethenc.json data/hadith_embeddings.npy data/hadith_embeddings_ids.json
from google.colab import files
files.download("/content/yaqeen-data.zip")

/content/Yaqeen/backend


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<div dir="rtl">

## 9. تركيب الإضافة في Chrome أو Edge (على جهازك، ليس في كولاب)
1. افتحي ملف `yaqeen-extension.zip` الذي نزل، واستخرجي ما فيه إلى مجلد (اضغطي عليه بالزر الأيمن ← Extract All).
2. في Chrome اكتبي في شريط العنوان: `chrome://extensions` (وفي Edge: `edge://extensions`).
3. فعّلي **Developer mode** (وضع المطوّر) من الزاوية.
4. اضغطي **Load unpacked** واختاري المجلد الذي استخرجتِه.
5. ستفتح صفحة إعدادات يقين: اضغطي **تنزيل البيانات الآن** وانتظري حتى يظهر عدد الآيات (6236) وعدد الأحاديث.
6. اضغطي أيقونة يقين بجانب شريط العنوان، والصقي آية أو حديثًا، واضغطي فحص.
7. افتحي X أو Facebook أو TikTok وسترين علامة يقين بجانب المنشورات التي فيها آيات أو أحاديث.
</div>